java/client/src/main/java/glide/connectors/handlers/ChannelHandler.java

```diff
@@ -84,6 +90,7 @@ public CompletableFuture<Response> connect(ConnectionRequest request) {
 
     /** Closes the UDS connection and frees corresponding resources. */
     public ChannelFuture close() {
+        this.isClosed.set(true);
```

Comment:
Why would it be easier? This is quite similar to what is being done in Py and Node close functions.

Reply:
it's not critical. just seems like it would simplify the implementation.
each language has its own affordances for socket usage. It seems like Java allows for good understanding of the state of the socket, but if there are still some scenarios where the socket isn't closed, use additional state.

Reply:
You can always add more state ;)